Modules/_typesmodule.c

```diff
@@ -1,17 +1,51 @@
 /* _types module */
 
 #include "Python.h"
+#include "pycore_descrobject.h"   // _PyMethodWrapper_Type
 #include "pycore_namespace.h"     // _PyNamespace_Type
+#include "pycore_object.h"       // _PyNone_Type, _PyNotImplemented_Type
+#include "pycore_unionobject.h"  // _PyUnion_Type
 
 static int
 _types_exec(PyObject *m)
 {
-    if (PyModule_AddObjectRef(m, "CapsuleType", (PyObject *)&PyCapsule_Type) < 0) {
-        return -1;
-    }
-    if (PyModule_AddObjectRef(m, "SimpleNamespace", (PyObject *)&_PyNamespace_Type) < 0) {
-        return -1;
-    }
+#define EXPORT_STATIC_TYPE(NAME, TYPE)                                   \
+    do {                                                                 \
+        if (PyModule_AddObjectRef(m, (NAME), (PyObject *)&(TYPE)) < 0) { \
```

Comment:
Let's make sure we're actually using static types:
```suggestion
        assert(PyUnstable_IsImmortal((PyObject *)&(TYPE)));
        if (PyModule_AddObjectRef(m, (NAME), (PyObject *)&(TYPE)) < 0) { \
```

(I think there is a `_Py_IsStaticImmortal` somewhere, but let's just use the public API here.)

Reply:
Good idea (applied manually to add a backslash for the macro)